### ISSUE TITLE
[a11y]: Menu component is not accessible with tab/shift tab navigation

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#16388]
### Package

@carbon/react

### Browser

Firefox

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

Menu component is not accessible with tab/shift tab navigation

### Recommended Fix
Make the menu tabbable using tabindex="0"

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-menu--playground#storybook-preview-wrapper

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
